lib/ruby_lsp/document.rb

```diff
@@ -63,9 +69,8 @@ def parse
 
       @tree = SyntaxTree.parse(@source)
       @unparsed_edits.clear
-      @parsable_source = @source.dup
-    rescue SyntaxTree::Parser::ParseError
-      update_parsable_source(@unparsed_edits)
+    rescue StandardError
```

Comment:
Do we want to suppress other errors?

Reply:
I'm also a bit worry about this biting us later in the future. Can we be more specific on the error class?

Reply:
I switched back to parser errors. The reason I had changed to `StandardError` is because we've been getting some `Comment location overlaps node location` errors, which are `RuntimeError` and notoriously difficult to reproduce. But indeed this may mask other problems, I agree.